streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -33,7 +33,7 @@ import org.apache.spark.streaming.Time
 @DeveloperApi
 case class BatchInfo(
     batchTime: Time,
-    streamIdToNumRecords: Map[Int, Long],
```

Comment:
This is a problem. We cannot expose `InputInfo` class through `BatchInfo` as the first one is private[streaming] whereas the latter in public. 

I wondering what is the right way here. Any ideas?